apricot/local/service.go

```diff
@@ -122,6 +122,23 @@ func (s *Service) GetDefaults() map[string]string {
 	return smap
 }
 
+func (s *Service) GetHostInventory(keyPrefix string) (hosts []string, err error) {
+	keys, err := s.src.GetKeysByPrefix(keyPrefix)
+	if err != nil {
+		log.WithError(err).Fatal("Error, could not retrieve host list.")
+		return []string{""}, err
```

Comment:
Please note that `[]string{""} != []string{}`.

The first one is a slice of strings with `len=1`, that contains a string with `len=0`. It is not an empty slice, so not quite a null value in this context.
The second one is a slice of strings with `len=0`, so an actual empty slice. This is what you should return.